### PR TITLE
[Feature/Action] Enable the popping of multiple scenes

### PR DESCRIPTION
**Pull request by rseemann**
Create the `popToParent()` Action that takes a parent name as argument and pops the scenes until the targeted parent is the current scene in place.

### CODE REVIEW

**Comment by rseemann** [comment]
@joenoon what do you think of this way of implementing the multiple pops?

**Comment by joenoon** [comment]
@rseemann cool!  This looks like it makes sense.  The one thing I'm unsure of is the state.children search - what if the child hierarchy is more complicated?  Does this need to walk the full tree instead of only direct children?  I'm not sure.  Would be awesome if you could add something to the Example app to demo the intended scenario - that would open up a place to experiment with more popToParent scenarios.

**Comment by rseemann** [comment]
@joenoon I also thought about a more complex structure, but I wouldn't know how to handle it, since I don't fully understand the routing structure. Maybe I'll do some testing with the Example app as you mentioned, great idea.

**Comment by aksonov** [comment]
@rseemann Route structure is very simple - it is just tree, with children as scenes. Child could have other children, etc.

**Comment by aksonov** [comment]
@rseemann could you also fix eslint errors?

**Comment by rseemann** [comment]
@aksonov The structure I kind of understand, popping those, instead, I'm not sure I understand, specially  with the index number. About the linting, I'll fix those, for sure.

**Comment by aksonov** [comment]
@rseemann index number is currently selected scene.

**Comment by rseemann** [comment]
@aksonov so this number does not depend on how deep the structure is, but rather on how many scenes there are currently in the stack?

**Comment by aksonov** [comment]
@rseemann yes. the same as in NavigationExperimental structure.

**Comment by aksonov** [comment]
For 'tabs', it represents number of selected tab.

**Comment by rseemann** [comment]
@aksonov that seems simple enough then, I'll make the necessary changes, add some examples and fix the linting then. Thanks for the explanations.

File: src/Reducer.js [hunk 1]
```diff
@@ -56,18 +57,22 @@ function inject(state, action, props, scenes) {
     return state;
   }
   let ind;
+  let numPops = 1;
```
Comment (Swordsman-Inaction) [inline src/Reducer.js]: Sorry, I didn't see yours before I create the PR #806 , I write something similar to your `numPops` . According this issue #39 , I think old version `Actions.pop()` could do `Actions.pop(2)` things, so I implemented this with several `assert` checking.

**Comment by rseemann** [comment]
@joenoon and @aksonov I made some much deeper changes to handle the navigation to any scene that exists in the stack, node or component. Both usages can be seen at the example, with the newly component login3. I'm not fully happy with the node/component detection, maybe you can tell me if that's a reliable check.

**Comment by BenBenz** [comment]
@rseemann  popToParent() works at this time ?

**Comment by rseemann** [comment]
@BenBenz I haven't checked on this since a while. Unless something changed in the navigation way, it should work.

**Comment by BenBenz** [comment]
@rseemann Thanks for this information 👍

**Pull request merged**